instrumentation-api-incubator/src/main/jflex/SqlSanitizer.jflex

```diff
@@ -122,13 +122,48 @@ WHITESPACE          = [ \t\r\n]+
     /** @return true if all statement info is gathered */
     boolean handleNext() {
       return false;
-     }
+    }
+
+    /** @return true if all statement info is gathered */
+    boolean handleOperationTarget(String target) {
+      return false;
+    }
+
+    boolean waitingOperationTarget() {
+      return false;
+    }
 
     SqlStatementInfo getResult(String fullStatement) {
       return SqlStatementInfo.create(fullStatement, getClass().getSimpleName().toUpperCase(java.util.Locale.ROOT), mainIdentifier);
     }
   }
 
+  private static abstract class CompositeOp extends Operation {
+    private String operationTarget = "";
+    private boolean waitingOperationTarget = true;
+
+    boolean waitingOperationTarget() {
+      return waitingOperationTarget;
+    }
+
+    boolean handleOperationTarget(String target) {
+      operationTarget = target;
+      waitingOperationTarget = false;
+      return false;
+    }
+
+    boolean shouldHandleIdentifier() {
+      return "TABLE".equals(operationTarget);
```

Comment:
I think you should add a comment here explaining why you chose to only capture the name for tables, it is not obvious for me.

Reply:
Added comment.